### ISSUE TITLE
Issue computing test values for SkLearn models

### ISSUE THREAD

**Issue by Qu3tzal** [ireydiak/anomaly_detection_NRCAN#15]
Because we iterate over the test set twice, we get two different orders (due to `DataLoader(..., shuffle=True, ...)`) which messes up the metrics computation.
First iteration over the test set happens here:
https://github.com/ireydiak/dagmm/blob/5e28dff979c659f49f7e9e7b4a7a886e3d4c168f/src/main.py#L271

Second iteration over the test set happens here:
https://github.com/ireydiak/dagmm/blob/5e28dff979c659f49f7e9e7b4a7a886e3d4c168f/src/main.py#L290-L292

We then compare the anomaly scores computed on the 2nd iterations to the truth values of the 1st iteration, however because each iteration provides batches in different orders, the metrics can't be computed properly resulting in poor results.
https://github.com/ireydiak/dagmm/blob/5e28dff979c659f49f7e9e7b4a7a886e3d4c168f/src/main.py#L300-L301

To solve we need to make sure `y_test` is in the same order as the values from `anomaly_score_test`.
@djeffkanda
